app/src/main/java/io/pslab/models/CompassData.java

```diff
@@ -0,0 +1,40 @@
+package io.pslab.models;
+
+public class CompassData {
+    private String Bx;
+    private String By;
+    private String Bz;
+
+    public CompassData() {
+    }
+
+    public CompassData(String x, String y, String z) {
```

Comment:
Use standard practice in object creation. We use the same name and use `this` key word to refer to the class instance. Change x y z to Bx By and Bz

Reply:
Updated. @CloudyPadmal